### PR TITLE
Second round of optimizations extracted from PR to implement `enum` variants

### PR DESCRIPTION
**Pull request by Kijewski**

### CODE REVIEW

File: rinja_derive/src/input.rs [hunk 1]
```diff
@@ -547,9 +547,9 @@ fn extension(path: &Path) -> Option<&str> {
     }
 }
 
-#[derive(Debug, Hash, PartialEq)]
+#[derive(Debug, Clone, Hash, PartialEq)]
 pub(crate) enum Source {
-    Path(String),
+    Path(Arc<str>),
```
Comment (GuillaumeGomez) [inline rinja_derive/src/input.rs:548]: Why an `Arc` and not an `Rc` btw? Do we need `Source` to be `Send`?

Comment (Kijewski) [reply rinja_derive/src/input.rs:548]: The other variant `Source::Source` needs an `Arc` to be compatible with cached file sources. `Source::Path` does not, but I think making both variants the same is still the better call. The Clone and Drop implementations are half the size, which I daresay outperforms any drawbacks the use of atomic numbers bring.

Comment (GuillaumeGomez) [reply rinja_derive/src/input.rs:548]: Right, forgot about the static. That answers my question. :)

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**